modules/core/src/test/java/org/apache/ignite/internal/processors/cache/transactions/TxWithKeyContentionSelfTest.java

```diff
@@ -286,50 +236,31 @@ private void runKeyCollisionsMetric(TransactionConcurrency concurrency, Transact
 
         finishFut.markInitialized();
 
-        for (Ignite ig0 : G.allGrids()) {
-            TestRecordingCommunicationSpi commSpi0 =
-                (TestRecordingCommunicationSpi)ig0.configuration().getCommunicationSpi();
-
-            if (ig0.configuration().isClientMode())
-                continue;
-
-            commSpi0.stopBlock();
-        }
-
-        IgniteTxManager srvTxMgr = ((IgniteEx)ig).context().cache().context().tm();
-
         assertTrue(GridTestUtils.waitForCondition(new GridAbsPredicate() {
             @Override public boolean apply() {
                 try {
```

Comment:
This try catch is actually useless, because exception is propagated to main thread. I suggest to add try-finally statement, with futures finishing in finally block.